### PR TITLE
Wait for prometheus to become healthy.

### PR DESCRIPTION
**Pull request by mm4tt**
We shouldn't run tests until the prometheus server is healthy and collects metrics from all targets.

https://github.com/kubernetes/kubernetes/issues/74213

### CODE REVIEW

File: clusterloader2/pkg/prometheus/prometheus.go [hunk 1]
```diff
@@ -99,3 +110,51 @@ func applyManifests(
...
+
+	var response targetsResponse
+	if err := json.Unmarshal(raw, &response); err != nil {
+		// This shouldn't happen, return error.
+		return false, err
+	}
+
+	for _, t := range response.Data.ActiveTargets {
+		if t.Health != "up" {
+			klog.Infof("Target {job=%s, pod=%s} not ready...", t.Labels["job"], t.Labels["pod"])
```
Comment (wojtek-t) [inline clusterloader2/pkg/prometheus/prometheus.go]: How does registering targets work?
Is it possible that we call this function immediately and there won't be any targets yet (and thus we satisfy it), and then the new (no yet up) will reigster?

File: clusterloader2/pkg/prometheus/prometheus.go [hunk 2]
```diff
@@ -99,3 +110,51 @@ func applyManifests(
...
+	klog.Info("Waiting for Prometheus stack to become healthy...")
+	return wait.Poll(
+		checkPrometheusReadyInterval,
+		checkPrometheusReadyTimeout,
+		func() (bool, error) { return isPrometheusReady(client) })
+}
+
+func isPrometheusReady(client clientset.Interface) (bool, error) {
+	raw, err := client.CoreV1().
+		Services("monitoring").
```
Comment (krzysied) [inline clusterloader2/pkg/prometheus/prometheus.go]: nit: You added const for namespace name.

File: clusterloader2/pkg/prometheus/prometheus.go (hunk 1 above)
Comment (mm4tt) [reply clusterloader2/pkg/prometheus/prometheus.go]: IIUC, they're created from the ServiceMonitor (CRD) objects and it's very unlikely that the prometheus server will see no targets - even though they might be created after the prometheus server deployment, it's quite unlikely that starting up prometheus server pod will outrun creating these objects. 

Nevertheless I added a condition to expect at least # nodes ready target. That should guarantee that we won't run into the situation you described.

File: clusterloader2/pkg/prometheus/prometheus.go (hunk 2 above)
Comment (mm4tt) [reply clusterloader2/pkg/prometheus/prometheus.go]: Thanks, done!

**Pull request merged**